src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.OSX.cs

```diff
@@ -18,40 +17,58 @@ public static int[] GetProcessIds()
 
         private static ProcessInfo CreateProcessInfo(int pid)
         {
-            ProcessInfo dummy = new ProcessInfo() // assign to fields to suppress warnings until implemented
+            // Negative PIDs aren't valid
+            if (pid < 0)
             {
-                _basePriority = 0,
-                _pageFileBytes = 0,
-                _handleCount = 0,
-                _pageFileBytesPeak = 0,
-                _poolNonpagedBytes = 0,
-                _poolPagedBytes = 0,
-                _privateBytes = 0,
-                _processId = pid,
-                _processName = null,
-                _sessionId = 0,
-                _virtualBytes = 0,
-                _virtualBytesPeak = 0,
-                _workingSet = 0,
-                _workingSetPeak = 0
-            };
-            dummy._threadInfoList.Add(new ThreadInfo()
+                throw new ArgumentOutOfRangeException("pid");
+            }
+
+            ProcessInfo procInfo = new ProcessInfo();
+
+            // Try to get the task info. This can fail if the user permissions don't permit
+            // this user context to query the specified process
+            Interop.libproc.proc_taskallinfo? info = Interop.libproc.GetProcessInfoById(pid);
+            if (info.HasValue == false)
             {
-                _basePriority = 0,
-                _currentPriority = 0,
-                _processId = pid,
-                _startAddress = IntPtr.Zero,
-                _threadId = 0,
-                _threadState = 0,
-                _threadWaitReason = ThreadWaitReason.Unknown
-            });
-
-            // TODO: Implement this
-            throw NotImplemented.ByDesign;
+                procInfo.BasePriority = info.Value.ptinfo.pti_priority;
+                procInfo.HandleCount = Interop.libproc.GetFileDescriptorsForPid(pid).Count;
+                procInfo.ProcessId = pid;
+                procInfo.ProcessName = info.Value.pbsd.pbi_comm;
+                procInfo.VirtualBytes = (long)info.Value.ptinfo.pti_virtual_size;
+                procInfo.WorkingSet = (long)info.Value.ptinfo.pti_resident_size;
+
+                // These values don't have meaning or don't exist on OSX
+                procInfo.PageFileBytes = -1;
```

Comment:
You're initializing these values to -1 in the ProcessInfo struct, so we don't need to do so here as well, right?